### ISSUE TITLE
Table of contents sidebar does not show all headings when first heading is not highest level

### ISSUE THREAD

**Issue by bsmith89** [jupyterlab/jupyterlab#14877]
## Description

The table of contents (TOC) displays incorrectly for some document structures.
Specifically, when the first markdown heading level is lower rank than one below it.

## Reproduce

1. Open a new notebook
2. Change the first cell to Markdown
3. Enter the markdown code as shown below.
4. Open the TOC panel on the left
5. Notice the set of headers displayed

A markdown cell with the following will generate the _expected_ TOC hierarchy:

```
# HEADER 1

# HEADER 2

## HEADER 2.1

### HEADER 2.1.1

# HEADER 3
```

![image](https://github.com/jupyterlab/jupyterlab/assets/711411/4c6c9d77-b0b0-455e-90bc-bc5cf057a6fc)

but the following (only changing the level of the first header) will be missing all but "HEADER 1" in the TOC:

```
## HEADER 1
^ See here

# HEADER 2

## HEADER 2.1

### HEADER 2.1.1

# HEADER 3
```

![image](https://github.com/jupyterlab/jupyterlab/assets/711411/a92b8b60-0c84-44c3-ba13-44bad80a7b54)

## Expected behavior

The TOC should look approximately the same as

```
# HEADER 1

# HEADER 2

## HEADER 2.1

### HEADER 2.1.1

# HEADER 3
```

but with the first header indented by one level.

## Context

- Operating System and version:
```
NAME="Red Hat Enterprise Linux Server"
VERSION="7.9 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.9"
PRETTY_NAME=RHEL
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.9:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.9
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.9"
```
- Browser and version: Version 114.0.5735.198 (Official Build) (x86_64)
- JupyterLab version: Version 4.0.3

Running JupyterLab on a remote server with SSH port tunneling.

<details><summary>Troubleshoot Output</summary>
<pre>
Singularity> jupyter troubleshoot
$PATH:
        $HOME/.snakemake/conda/27f7180629009d6ab586849a4a3ef3f7_/bin
        /opt/conda/bin
        /usr/local/sbin
        /usr/local/bin
        /usr/sbin
        /usr/bin
        /sbin
        /bin

sys.path:
        $HOME/.snakemake/conda/27f7180629009d6ab586849a4a3ef3f7_/bin
        $HOME/.snakemake/conda/27f7180629009d6ab586849a4a3ef3f7_/lib/python310.zip
        $HOME/.snakemake/conda/27f7180629009d6ab586849a4a3ef3f7_/lib/python3.10
        $HOME/.snakemake/conda/27f7180629009d6ab586849a4a3ef3f7_/lib/python3.10/lib-dynload
        $HOME/.snakemake/conda/27f7180629009d6ab586849a4a3ef3f7_/lib/python3.10/site-packages

sys.executable:
        $HOME/.snakemake/conda/27f7180629009d6ab586849a4a3ef3f7_/bin/python

sys.version:
        3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:40:32) [GCC 12.3.0]

platform.platform():
        Linux-3.10.0-1160.76.1.el7.x86_64-x86_64-with-glibc2.31

which -a jupyter:
        $HOME/.snakemake/conda/27f7180629009d6ab586849a4a3ef3f7_/bin/jupyter
        /opt/conda/bin/jupyter

pip list:
        Package                       Version
        ----------------------------- -----------
        anyio                         3.7.1
        argon2-cffi                   21.3.0
        argon2-cffi-bindings          21.2.0
        arrow                         1.2.3
        asttokens                     2.2.1
        async-lru                     2.0.3
        attrs                         23.1.0
        Babel                         2.12.1
        backcall                      0.2.0
        backports.functools-lru-cache 1.6.5
        beautifulsoup4                4.12.2
        black                         23.7.0
        bleach                        6.0.0
        Brotli                        1.0.9
        cached-property               1.5.2
        certifi                       2023.7.22
        cffi                          1.15.1
        charset-normalizer            3.2.0
        click                         8.1.6
        comm                          0.1.3
        debugpy                       1.6.7
        decorator                     5.1.1
        defusedxml                    0.7.1
        docstring-to-markdown         0.12
        entrypoints                   0.4
        exceptiongroup                1.1.2
        executing                     1.2.0
        fastjsonschema                2.18.0
        flit_core                     3.9.0
        fqdn                          1.5.1
        idna                          3.4
        importlib-metadata            6.8.0
        importlib-resources           6.0.0
        ipykernel                     6.25.0
        ipython                       8.14.0
        isoduration                   20.11.0
        jedi                          0.18.0
        jedi-language-server          0.34.8
        Jinja2                        3.1.2
        json5                         0.9.14
        jsonpointer                   2.0
        jsonschema                    4.18.4
        jsonschema-specifications     2023.7.1
        jupyter_client                8.3.0
        jupyter_core                  5.3.1
        jupyter-events                0.6.3
        jupyter-lsp                   2.2.0
        jupyter_server                2.7.0
        jupyter_server_terminals      0.4.4
        jupyterlab                    4.0.3
        jupyterlab_code_formatter     2.2.1
        jupyterlab-pygments           0.2.2
        jupyterlab_server             2.24.0
        MarkupSafe                    2.1.3
        matplotlib-inline             0.1.6
        mistune                       3.0.0
        mypy-extensions               1.0.0
        nbclient                      0.8.0
        nbconvert                     7.7.3
        nbformat                      5.9.1
        nest-asyncio                  1.5.6
        notebook                      7.0.0
        notebook_shim                 0.2.3
        overrides                     7.3.1
        packaging                     23.1
        pandocfilters                 1.5.0
        parso                         0.8.3
        pathspec                      0.11.1
        pexpect                       4.8.0
        pickleshare                   0.7.5
        pip                           23.2.1
        pkgutil_resolve_name          1.3.10
        platformdirs                  3.9.1
        prometheus-client             0.17.1
        prompt-toolkit                3.0.39
        psutil                        5.9.5
        ptyprocess                    0.7.0
        pure-eval                     0.2.2
        pycparser                     2.21
        pydantic                      1.8.2
        pygls                         0.11.3
        Pygments                      2.15.1
        PySocks                       1.7.1
        python-dateutil               2.8.2
        python-json-logger            2.0.7
        pytz                          2023.3
        PyYAML                        6.0
        pyzmq                         25.1.0
        referencing                   0.30.0
        requests                      2.31.0
        rfc3339-validator             0.1.4
        rfc3986-validator             0.1.1
        rpds-py                       0.9.2
        Send2Trash                    1.8.2
        setuptools                    68.0.0
        six                           1.16.0
        sniffio                       1.3.0
        soupsieve                     2.3.2.post1
        stack-data                    0.6.2
        terminado                     0.17.1
        tinycss2                      1.2.1
        tomli                         2.0.1
        tornado                       6.3.2
        traitlets                     5.9.0
        typeguard                     2.13.3
        typing_extensions             4.7.1
        typing-utils                  0.1.0
        uri-template                  1.3.0
        urllib3                       2.0.4
        wcwidth                       0.2.6
        webcolors                     1.13
        webencodings                  0.5.1
        websocket-client              1.6.1
        wheel                         0.41.0
        zipp                          3.16.2

conda list:
        # packages in environment at $HOME/.snakemake/conda/27f7180629009d6ab586849a4a3ef3f7_:
        #
        # Name                    Version                   Build  Channel
        _libgcc_mutex             0.1                 conda_forge    conda-forge
        _openmp_mutex             4.5                       2_gnu    conda-forge
        anyio                     3.7.1              pyhd8ed1ab_0    conda-forge
        argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
        argon2-cffi-bindings      21.2.0          py310h5764c6d_3    conda-forge
        arrow                     1.2.3              pyhd8ed1ab_0    conda-forge
        asttokens                 2.2.1              pyhd8ed1ab_0    conda-forge
        async-lru                 2.0.3              pyhd8ed1ab_0    conda-forge
        attrs                     23.1.0             pyh71513ae_1    conda-forge
        babel                     2.12.1             pyhd8ed1ab_1    conda-forge
        backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
        backports                 1.0                pyhd8ed1ab_3    conda-forge
        backports.functools_lru_cache 1.6.5              pyhd8ed1ab_0    conda-forge
        beautifulsoup4            4.12.2             pyha770c72_0    conda-forge
        black                     23.7.0          py310hff52083_1    conda-forge
        bleach                    6.0.0              pyhd8ed1ab_0    conda-forge
        brotli-python             1.0.9           py310hd8f1fbe_9    conda-forge
        bzip2                     1.0.8                h7f98852_4    conda-forge
        ca-certificates           2023.7.22            hbcca054_0    conda-forge
        cached-property           1.5.2                hd8ed1ab_1    conda-forge
        cached_property           1.5.2              pyha770c72_1    conda-forge
        certifi                   2023.7.22          pyhd8ed1ab_0    conda-forge
        cffi                      1.15.1          py310h255011f_3    conda-forge
        charset-normalizer        3.2.0              pyhd8ed1ab_0    conda-forge
        click                     8.1.6           unix_pyh707e725_0    conda-forge
        comm                      0.1.3              pyhd8ed1ab_0    conda-forge
        debugpy                   1.6.7           py310heca2aa9_0    conda-forge
        decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
        defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
        docstring-to-markdown     0.12               pyhd8ed1ab_0    conda-forge
        entrypoints               0.4                pyhd8ed1ab_0    conda-forge
        exceptiongroup            1.1.2              pyhd8ed1ab_0    conda-forge
        executing                 1.2.0              pyhd8ed1ab_0    conda-forge
        flit-core                 3.9.0              pyhd8ed1ab_0    conda-forge
        fqdn                      1.5.1              pyhd8ed1ab_0    conda-forge
        idna                      3.4                pyhd8ed1ab_0    conda-forge
        importlib-metadata        6.8.0              pyha770c72_0    conda-forge
        importlib_metadata        6.8.0                hd8ed1ab_0    conda-forge
        importlib_resources       6.0.0              pyhd8ed1ab_1    conda-forge
        ipykernel                 6.25.0             pyh71e2992_0    conda-forge
        ipython                   8.14.0             pyh41d4057_0    conda-forge
        isoduration               20.11.0            pyhd8ed1ab_0    conda-forge
        jedi                      0.18.0          py310hff52083_3    conda-forge
        jedi-language-server      0.34.8             pyhd8ed1ab_0    conda-forge
        jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
        json5                     0.9.14             pyhd8ed1ab_0    conda-forge
        jsonpointer               2.0                        py_0    conda-forge
        jsonschema                4.18.4             pyhd8ed1ab_0    conda-forge
        jsonschema-specifications 2023.7.1           pyhd8ed1ab_0    conda-forge
        jsonschema-with-format-nongpl 4.18.4             pyhd8ed1ab_0    conda-forge
        jupyter-lsp               2.2.0              pyhd8ed1ab_0    conda-forge
        jupyter_client            8.3.0              pyhd8ed1ab_0    conda-forge
        jupyter_core              5.3.1           py310hff52083_0    conda-forge
        jupyter_events            0.6.3              pyhd8ed1ab_1    conda-forge
        jupyter_server            2.7.0              pyhd8ed1ab_0    conda-forge
        jupyter_server_terminals  0.4.4              pyhd8ed1ab_1    conda-forge
        jupyterlab                4.0.3              pyhd8ed1ab_0    conda-forge
        jupyterlab_code_formatter 2.2.1              pyhd8ed1ab_0    conda-forge
        jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
        jupyterlab_server         2.24.0             pyhd8ed1ab_0    conda-forge
        ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
        libffi                    3.4.2                h7f98852_5    conda-forge
        libgcc-ng                 13.1.0               he5830b7_0    conda-forge
        libgomp                   13.1.0               he5830b7_0    conda-forge
        libnsl                    2.0.0                h7f98852_0    conda-forge
        libsodium                 1.0.18               h36c2ea0_1    conda-forge
        libsqlite                 3.42.0               h2797004_0    conda-forge
        libstdcxx-ng              13.1.0               hfd8a6a1_0    conda-forge
        libuuid                   2.38.1               h0b41bf4_0    conda-forge
        libzlib                   1.2.13               hd590300_5    conda-forge
        markupsafe                2.1.3           py310h2372a71_0    conda-forge
        matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
        mistune                   3.0.0              pyhd8ed1ab_0    conda-forge
        mypy_extensions           1.0.0              pyha770c72_0    conda-forge
        nbclient                  0.8.0              pyhd8ed1ab_0    conda-forge
        nbconvert-core            7.7.3              pyhd8ed1ab_0    conda-forge
        nbformat                  5.9.1              pyhd8ed1ab_0    conda-forge
        ncurses                   6.4                  hcb278e6_0    conda-forge
        nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
        notebook                  7.0.0              pyhd8ed1ab_0    conda-forge
        notebook-shim             0.2.3              pyhd8ed1ab_0    conda-forge
        openssl                   3.1.1                hd590300_1    conda-forge
        overrides                 7.3.1              pyhd8ed1ab_0    conda-forge
        packaging                 23.1               pyhd8ed1ab_0    conda-forge
        pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
        parso                     0.8.3              pyhd8ed1ab_0    conda-forge
        pathspec                  0.11.1             pyhd8ed1ab_0    conda-forge
        pexpect                   4.8.0              pyh1a96a4e_2    conda-forge
        pickleshare               0.7.5                   py_1003    conda-forge
        pip                       23.2.1             pyhd8ed1ab_0    conda-forge
        pkgutil-resolve-name      1.3.10             pyhd8ed1ab_0    conda-forge
        platformdirs              3.9.1              pyhd8ed1ab_0    conda-forge
        prometheus_client         0.17.1             pyhd8ed1ab_0    conda-forge
        prompt-toolkit            3.0.39             pyha770c72_0    conda-forge
        prompt_toolkit            3.0.39               hd8ed1ab_0    conda-forge
        psutil                    5.9.5           py310h1fa729e_0    conda-forge
        ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
        pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
        pycparser                 2.21               pyhd8ed1ab_0    conda-forge
        pydantic                  1.8.2           py310h6acc77f_2    conda-forge
        pygls                     0.11.3             pyhd8ed1ab_0    conda-forge
        pygments                  2.15.1             pyhd8ed1ab_0    conda-forge
        pysocks                   1.7.1              pyha2e5f31_6    conda-forge
        python                    3.10.12         hd12c33a_0_cpython    conda-forge
        python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
        python-fastjsonschema     2.18.0             pyhd8ed1ab_0    conda-forge
        python-json-logger        2.0.7              pyhd8ed1ab_0    conda-forge
        python_abi                3.10                    3_cp310    conda-forge
        pytz                      2023.3             pyhd8ed1ab_0    conda-forge
        pyyaml                    6.0             py310h5764c6d_5    conda-forge
        pyzmq                     25.1.0          py310h5bbb5d0_0    conda-forge
        readline                  8.2                  h8228510_1    conda-forge
        referencing               0.30.0             pyhd8ed1ab_0    conda-forge
        requests                  2.31.0             pyhd8ed1ab_0    conda-forge
        rfc3339-validator         0.1.4              pyhd8ed1ab_0    conda-forge
        rfc3986-validator         0.1.1              pyh9f0ad1d_0    conda-forge
        rpds-py                   0.9.2           py310hcb5633a_0    conda-forge
        send2trash                1.8.2              pyh41d4057_0    conda-forge
        setuptools                68.0.0             pyhd8ed1ab_0    conda-forge
        six                       1.16.0             pyh6c4a22f_0    conda-forge
        sniffio                   1.3.0              pyhd8ed1ab_0    conda-forge
        soupsieve                 2.3.2.post1        pyhd8ed1ab_0    conda-forge
        stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
        terminado                 0.17.1             pyh41d4057_0    conda-forge
        tinycss2                  1.2.1              pyhd8ed1ab_0    conda-forge
        tk                        8.6.12               h27826a3_0    conda-forge
        tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
        tornado                   6.3.2           py310h2372a71_0    conda-forge
        traitlets                 5.9.0              pyhd8ed1ab_0    conda-forge
        typeguard                 2.13.3             pyhd8ed1ab_0    conda-forge
        typing-extensions         4.7.1                hd8ed1ab_0    conda-forge
        typing_extensions         4.7.1              pyha770c72_0    conda-forge
        typing_utils              0.1.0              pyhd8ed1ab_0    conda-forge
        tzdata                    2023c                h71feb2d_0    conda-forge
        uri-template              1.3.0              pyhd8ed1ab_0    conda-forge
        urllib3                   2.0.4              pyhd8ed1ab_0    conda-forge
        wcwidth                   0.2.6              pyhd8ed1ab_0    conda-forge
        webcolors                 1.13               pyhd8ed1ab_0    conda-forge
        webencodings              0.5.1                      py_1    conda-forge
        websocket-client          1.6.1              pyhd8ed1ab_0    conda-forge
        wheel                     0.41.0             pyhd8ed1ab_0    conda-forge
        xz                        5.2.6                h166bdaf_0    conda-forge
        yaml                      0.2.5                h7f98852_2    conda-forge
        zeromq                    4.3.4                h9c3ff4c_1    conda-forge
        zipp                      3.16.2             pyhd8ed1ab_0    conda-forge

conda env:
        name: $HOME/.snakemake/conda/27f7180629009d6ab586849a4a3ef3f7_
        channels:
          - conda-forge
        dependencies:
          - _libgcc_mutex=0.1=conda_forge
          - _openmp_mutex=4.5=2_gnu
          - anyio=3.7.1=pyhd8ed1ab_0
          - argon2-cffi=21.3.0=pyhd8ed1ab_0
          - argon2-cffi-bindings=21.2.0=py310h5764c6d_3
          - arrow=1.2.3=pyhd8ed1ab_0
          - asttokens=2.2.1=pyhd8ed1ab_0
          - async-lru=2.0.3=pyhd8ed1ab_0
          - attrs=23.1.0=pyh71513ae_1
          - babel=2.12.1=pyhd8ed1ab_1
          - backcall=0.2.0=pyh9f0ad1d_0
          - backports=1.0=pyhd8ed1ab_3
          - backports.functools_lru_cache=1.6.5=pyhd8ed1ab_0
          - beautifulsoup4=4.12.2=pyha770c72_0
          - black=23.7.0=py310hff52083_1
          - bleach=6.0.0=pyhd8ed1ab_0
          - brotli-python=1.0.9=py310hd8f1fbe_9
          - bzip2=1.0.8=h7f98852_4
          - ca-certificates=2023.7.22=hbcca054_0
          - cached-property=1.5.2=hd8ed1ab_1
          - cached_property=1.5.2=pyha770c72_1
          - certifi=2023.7.22=pyhd8ed1ab_0
          - cffi=1.15.1=py310h255011f_3
          - charset-normalizer=3.2.0=pyhd8ed1ab_0
          - click=8.1.6=unix_pyh707e725_0
          - comm=0.1.3=pyhd8ed1ab_0
          - debugpy=1.6.7=py310heca2aa9_0
          - decorator=5.1.1=pyhd8ed1ab_0
          - defusedxml=0.7.1=pyhd8ed1ab_0
          - docstring-to-markdown=0.12=pyhd8ed1ab_0
          - entrypoints=0.4=pyhd8ed1ab_0
          - exceptiongroup=1.1.2=pyhd8ed1ab_0
          - executing=1.2.0=pyhd8ed1ab_0
          - flit-core=3.9.0=pyhd8ed1ab_0
          - fqdn=1.5.1=pyhd8ed1ab_0
          - idna=3.4=pyhd8ed1ab_0
          - importlib-metadata=6.8.0=pyha770c72_0
          - importlib_metadata=6.8.0=hd8ed1ab_0
          - importlib_resources=6.0.0=pyhd8ed1ab_1
          - ipykernel=6.25.0=pyh71e2992_0
          - ipython=8.14.0=pyh41d4057_0
          - isoduration=20.11.0=pyhd8ed1ab_0
          - jedi=0.18.0=py310hff52083_3
          - jedi-language-server=0.34.8=pyhd8ed1ab_0
          - jinja2=3.1.2=pyhd8ed1ab_1
          - json5=0.9.14=pyhd8ed1ab_0
          - jsonpointer=2.0=py_0
          - jsonschema=4.18.4=pyhd8ed1ab_0
          - jsonschema-specifications=2023.7.1=pyhd8ed1ab_0
          - jsonschema-with-format-nongpl=4.18.4=pyhd8ed1ab_0
          - jupyter-lsp=2.2.0=pyhd8ed1ab_0
          - jupyter_client=8.3.0=pyhd8ed1ab_0
          - jupyter_core=5.3.1=py310hff52083_0
          - jupyter_events=0.6.3=pyhd8ed1ab_1
          - jupyter_server=2.7.0=pyhd8ed1ab_0
          - jupyter_server_terminals=0.4.4=pyhd8ed1ab_1
          - jupyterlab=4.0.3=pyhd8ed1ab_0
          - jupyterlab_code_formatter=2.2.1=pyhd8ed1ab_0
          - jupyterlab_pygments=0.2.2=pyhd8ed1ab_0
          - jupyterlab_server=2.24.0=pyhd8ed1ab_0
          - ld_impl_linux-64=2.40=h41732ed_0
          - libffi=3.4.2=h7f98852_5
          - libgcc-ng=13.1.0=he5830b7_0
          - libgomp=13.1.0=he5830b7_0
          - libnsl=2.0.0=h7f98852_0
          - libsodium=1.0.18=h36c2ea0_1
          - libsqlite=3.42.0=h2797004_0
          - libstdcxx-ng=13.1.0=hfd8a6a1_0
          - libuuid=2.38.1=h0b41bf4_0
          - libzlib=1.2.13=hd590300_5
          - markupsafe=2.1.3=py310h2372a71_0
          - matplotlib-inline=0.1.6=pyhd8ed1ab_0
          - mistune=3.0.0=pyhd8ed1ab_0
          - mypy_extensions=1.0.0=pyha770c72_0
          - nbclient=0.8.0=pyhd8ed1ab_0
          - nbconvert-core=7.7.3=pyhd8ed1ab_0
          - nbformat=5.9.1=pyhd8ed1ab_0
          - ncurses=6.4=hcb278e6_0
          - nest-asyncio=1.5.6=pyhd8ed1ab_0
          - notebook=7.0.0=pyhd8ed1ab_0
          - notebook-shim=0.2.3=pyhd8ed1ab_0
          - openssl=3.1.1=hd590300_1
          - overrides=7.3.1=pyhd8ed1ab_0
          - packaging=23.1=pyhd8ed1ab_0
          - pandocfilters=1.5.0=pyhd8ed1ab_0
          - parso=0.8.3=pyhd8ed1ab_0
          - pathspec=0.11.1=pyhd8ed1ab_0
          - pexpect=4.8.0=pyh1a96a4e_2
          - pickleshare=0.7.5=py_1003
          - pip=23.2.1=pyhd8ed1ab_0
          - pkgutil-resolve-name=1.3.10=pyhd8ed1ab_0
          - platformdirs=3.9.1=pyhd8ed1ab_0
          - prometheus_client=0.17.1=pyhd8ed1ab_0
          - prompt-toolkit=3.0.39=pyha770c72_0
          - prompt_toolkit=3.0.39=hd8ed1ab_0
          - psutil=5.9.5=py310h1fa729e_0
          - ptyprocess=0.7.0=pyhd3deb0d_0
          - pure_eval=0.2.2=pyhd8ed1ab_0
          - pycparser=2.21=pyhd8ed1ab_0
          - pydantic=1.8.2=py310h6acc77f_2
          - pygls=0.11.3=pyhd8ed1ab_0
          - pygments=2.15.1=pyhd8ed1ab_0
          - pysocks=1.7.1=pyha2e5f31_6
          - python=3.10.12=hd12c33a_0_cpython
          - python-dateutil=2.8.2=pyhd8ed1ab_0
          - python-fastjsonschema=2.18.0=pyhd8ed1ab_0
          - python-json-logger=2.0.7=pyhd8ed1ab_0
          - python_abi=3.10=3_cp310
          - pytz=2023.3=pyhd8ed1ab_0
          - pyyaml=6.0=py310h5764c6d_5
          - pyzmq=25.1.0=py310h5bbb5d0_0
          - readline=8.2=h8228510_1
          - referencing=0.30.0=pyhd8ed1ab_0
          - requests=2.31.0=pyhd8ed1ab_0
          - rfc3339-validator=0.1.4=pyhd8ed1ab_0
          - rfc3986-validator=0.1.1=pyh9f0ad1d_0
          - rpds-py=0.9.2=py310hcb5633a_0
          - send2trash=1.8.2=pyh41d4057_0
          - setuptools=68.0.0=pyhd8ed1ab_0
          - six=1.16.0=pyh6c4a22f_0
          - sniffio=1.3.0=pyhd8ed1ab_0
          - soupsieve=2.3.2.post1=pyhd8ed1ab_0
          - stack_data=0.6.2=pyhd8ed1ab_0
          - terminado=0.17.1=pyh41d4057_0
          - tinycss2=1.2.1=pyhd8ed1ab_0
          - tk=8.6.12=h27826a3_0
          - tomli=2.0.1=pyhd8ed1ab_0
          - tornado=6.3.2=py310h2372a71_0
          - traitlets=5.9.0=pyhd8ed1ab_0
          - typeguard=2.13.3=pyhd8ed1ab_0
          - typing-extensions=4.7.1=hd8ed1ab_0
          - typing_extensions=4.7.1=pyha770c72_0
          - typing_utils=0.1.0=pyhd8ed1ab_0
          - tzdata=2023c=h71feb2d_0
          - uri-template=1.3.0=pyhd8ed1ab_0
          - urllib3=2.0.4=pyhd8ed1ab_0
          - wcwidth=0.2.6=pyhd8ed1ab_0
          - webcolors=1.13=pyhd8ed1ab_0
          - webencodings=0.5.1=py_1
          - websocket-client=1.6.1=pyhd8ed1ab_0
          - wheel=0.41.0=pyhd8ed1ab_0
          - xz=5.2.6=h166bdaf_0
          - yaml=0.2.5=h7f98852_2
          - zeromq=4.3.4=h9c3ff4c_1
          - zipp=3.16.2=pyhd8ed1ab_0
        prefix: $HOME/.snakemake/conda/27f7180629009d6ab586849a4a3ef3f7_

</pre>
</details>

<details><summary>Browser Output</summary>

<pre>
(Nothing shows up in the JS console when toggling between the broken and working examples.)
</pre>
</details>

**Comment by krassowski** [comment]
This appears to be a duplicate of #14574, should be fixed by #14753 (which will be included in JupyterLab 4.0.4).

**Comment by bsmith89** [comment]
Ah, apologies. My search for existing issues was apparently less thorough than I thought. 😬
